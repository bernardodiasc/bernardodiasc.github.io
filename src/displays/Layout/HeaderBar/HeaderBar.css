.Layout-HeaderBar {
  font-size: var(--text-size);
  display: flex;
  flex-direction: column;
  align-items: baseline;
  position: relative;
  color: white;
}

.Layout-HeaderBar a,
.Layout-HeaderBar a:visited {
  color: inherit;
}

.Layout-HeaderBar__title {
  color: inherit;
  font-size: var(--heading-size);
  margin: 0;
}

.Layout-HeaderBar__social {
  font-size: 0.7em;
}

.Layout-HeaderBar__social:hover::after {
  content: 'â¤´';
  position: absolute;
}

.Layout-HeaderBar__languages {
  margin: 0.5em 0 1em;
}
