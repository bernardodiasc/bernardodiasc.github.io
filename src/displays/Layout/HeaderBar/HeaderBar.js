import React, { PureComponent } from 'react'
import './HeaderBar.css'

import AppLink from 'displays/AppLink'

class HeaderBar extends PureComponent {
  static defaultProps = {
    languages: null,
  }

  render() {
    return (
      <header className="Layout-HeaderBar">
        <h1 className="Layout-HeaderBar__title">
          <AppLink>Random Thoughts</AppLink>
        </h1>
        {this.props.languages && (
          <div className="Layout-HeaderBar__languages">
            {this.props.languages}
          </div>
        )}
        <a
          className="Layout-HeaderBar__social"
          href="https://twitter.com/bernardodiasc"
          target="_blank"
          rel="noopener noreferrer"
        >
          Twitter
        </a>
        <a
          className="Layout-HeaderBar__social"
          href="https://github.com/bernardodiasc"
          target="_blank"
          rel="noopener noreferrer"
        >
          GitHub
        </a>
      </header>
    )
  }
}

export default HeaderBar
