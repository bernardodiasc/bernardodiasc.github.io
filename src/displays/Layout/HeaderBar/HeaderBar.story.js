import React from 'react'
import { storiesOf } from '@storybook/react'
import { withKnobs } from '@storybook/addon-knobs/react'
import Docs from 'stories/containers/Docs'

import HeaderBar from './HeaderBar'
import HeaderBarReadme from './README.md'
storiesOf('Displays/Layout/HeaderBar', module)
  .addDecorator(withKnobs)
  .add('Info', () => <Docs>{HeaderBarReadme}</Docs>)
  .add('Example', () => <HeaderBar />)
  .add('Minimal render', () => <HeaderBar />)
