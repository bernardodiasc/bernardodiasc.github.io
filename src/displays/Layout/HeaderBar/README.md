# HeaderBar

## How to use

```
import HeaderBar from 'displays/Layout/HeaderBar'
```

```
<HeaderBar />
```

## Properties

| propName | propType | defaultValue | isRequired |
| - | - | - | - |
