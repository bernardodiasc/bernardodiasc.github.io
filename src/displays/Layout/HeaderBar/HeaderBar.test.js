import React from 'react'
import { shallow } from 'enzyme'

import HeaderBar from './HeaderBar.js'

it('renders without crashing', () => {
  const component = shallow(<HeaderBar />)
  expect(component).toMatchSnapshot()
})
