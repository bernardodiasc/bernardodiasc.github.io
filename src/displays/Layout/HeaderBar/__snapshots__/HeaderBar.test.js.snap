// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<header
  className="Layout-HeaderBar"
>
  <h1
    className="Layout-HeaderBar__title"
  >
    <AppLink
      to=""
    >
      Random Thoughts
    </AppLink>
  </h1>
  <div
    className="Layout-HeaderBar__about"
  >
    <img
      alt="Bernardo Dias da Cruz"
      className="Layout-HeaderBar__about-picture"
      src="/images/me.jpg"
    />
    <p>
      I'm 
      <strong>
        Bernardo Dias da Cruz
      </strong>
      ,
      <br />
      a senior web developer.
    </p>
    <p>
      <AppLink
        to="about-me"
      >
        Click here to know more about me.
      </AppLink>
    </p>
    <p
      className="Layout-HeaderBar__social"
    >
      <span>
        Follow me on 
      </span>
      <a
        href="https://github.com/bernardodiasc"
        rel="noopener noreferrer"
        target="_blank"
      >
        GitHub
      </a>
      <span>
         and 
      </span>
      <a
        href="https://twitter.com/bernardodiasc"
        rel="noopener noreferrer"
        target="_blank"
      >
        Twitter
      </a>
    </p>
  </div>
  <nav
    className="Layout-HeaderBar__further"
  >
    <h2>
      Looking further?
    </h2>
    <ul
      className="Layout-HeaderBar__further-list"
    >
      <li
        className="Layout-HeaderBar__further-item"
      >
        <AppLink
          to="archive"
        >
          Archive
        </AppLink>
      </li>
      <li
        className="Layout-HeaderBar__further-item"
      >
        <AppLink
          to="search"
        >
          Search for content
        </AppLink>
      </li>
    </ul>
  </nav>
</header>
`;
